frontend/server/minio-helper.ts

```diff
@@ -12,7 +12,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 import { Stream } from 'stream';
-import * as tar from 'tar';
+import * as tar from 'tar-stream';
```

Comment:
`tar` couples the deflating/gunzip with extracting the tarball.

I need to search for the issue. Essentially, there is a slightly inconsistency in storing the artifact with Argo in diff storage backend. Some had gzip compression, others dun.

i.e. I recalled I had a bit of issue getting my metadata UI to render properly, because s3 artifact is stored differently from a minio artifact.

So there was a discussion on turning gzip off for everything to ensure consistency. So I need to switch to `tar-stream` and `maybe-gunzip` to handle compressed and uncompressed tarballs.

Let me search for the issue again. It was many months ago.

Reply:
Ok. I think I recalled more of it. This PR still haven't fix the issue. What we need is a `maybe tarball.`

U notice s3 artifacts are not in archived. While all minio artifacts are tarballs. 

In the past, before I added update the minio client to be able to retrieve from s3 w/o API key. For metadata-ui, we store the data in s3 in our kubeflow op. But retrieve in UI as a minio artifact (with a minio gateway to s3).

However, the API for minio artifact expects a tarball, but in reality our artifact is not.